extension/sigv4authextension/README.md

```diff
@@ -14,9 +14,10 @@ The configuration fields are as follows:
 * `assume_role`: Specifies the configuration needed to assume a role
   * `arn`: **Optional**. The Amazon Resource Name (ARN) of a role to assume
   * `session_name`: **Optional**. The name of a role session
-* `region`: **Optional**. The AWS region for AWS Sigv4
+  * `sts_region`: **Optional**. The AWS region where STS is used to assumed the configured role
```

Comment:
Isn't this required?  If `assume_role` is used then the STS region is required.

Reply:
There's a note under that says it's required when a role to assume is provided - but if not provided, it's optional in that case

Reply:
I guess there's no way to provide `assume_role::sts_region` without also providing `assume_role`.  It seems potentially confusing to say something is optional when it must be provided any time it can be provided.  Perhaps move the `optional` designation to `assume_role` and remove it from `assume_role::arn` and `assume_role::sts_region`.  The `assume_role::session_name` value is indeed optional, even if `assume_role` is present.